MusicNotationKit/MusicNotationKit/Clef.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  Clef.swift
+//  MusicNotationKit
+//
+//  Created by Kyle Sherman on 10/16/16.
+//  Copyright © 2016 Kyle Sherman. All rights reserved.
+//
+
+public struct Clef {
+
+    public let tone: Tone?
+    /**
+     Starts from 0 on the first line (from the bottom). Ledger lines below that are negative.
+     Each increase by 1 moves a half step. i.e. 1 is the first space on the staff.
+     */
+    internal var halfSteps: Int {
+        switch staffLocation.locationType {
+        case .space:
+            return staffLocation.number * 2 + 1
+        case .line:
+            return staffLocation.number * 2
+        }
+    }
+
+    public let staffLocation: StaffLocation
+
+    /**
+     You can create a custom clef by providing a tone and line number.
+
+     - parameter tone: The tone that the clef represents. Tone is optional to support un-pitched (i.e. drums)
+     - parameter location: The location on the staff
+     */
+    public init(tone: Tone?, location: StaffLocation) {
+        self.tone = tone
+        self.staffLocation = location
+    }
+
+    public static let treble = Clef(tone: Tone(noteLetter: .g, octave: .octave4), location: StaffLocation(type: .space, number: 1))
```

Comment:
True. I don't know what happened. I tried to convert and I guess I messed up. I'll fix all of these.